File: deal-mgmt/app/schemas/transcription.py
"""녹음 변환 — 스키마."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class TranscriptionJobOut(BaseModel):
    """변환 작업 상태/결과 응답."""

    id: uuid.UUID
    transaction_id: uuid.UUID
    title: str
    meeting_date: str
    meeting_phase: str
    buyer_id: uuid.UUID | None = None
    status: str
    transcript: str | None = None
    minutes_json: dict | None = None
    meeting_log_id: uuid.UUID | None = None
    stt_cost_krw: float = 0.0
    llm_cost_usd: float = 0.0
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TranscriptionApproval(BaseModel):
    """사용자가 결과를 검토/편집 후 확정."""

    minutes: str = Field(..., min_length=1, description="확정 회의록 텍스트")
    summary: str | None = Field(default=None, description="회의 요약")
    action_items: list[ActionItemInput] | None = None
    condition_match: str | None = None
    condition_notes: str | None = None
    buyer_reaction: str | None = None


class ActionItemInput(BaseModel):
    """액션아이템 입력."""

    title: str = Field(..., min_length=1)
    description: str | None = None
    assignee_name: str | None = None
    assignee_email: str | None = None
    due_date: str | None = None
    priority: str = "MEDIUM"


# Forward reference 해결
TranscriptionApproval.model_rebuild()

File: amic-platform/src/modules/ma/types/transcription.ts
// 녹음 변환 — Clova STT + LLM 자동 회의록

import type { MeetingPhase, BuyerReaction, ConditionMatchLevel } from "./meeting_log";

export type TranscriptionJobStatus =
  | "PENDING"
  | "TRANSCRIBING"
  | "ANALYZING"
  | "COMPLETED"
  | "APPROVED"
  | "FAILED";

export interface TranscriptionJob {
  id: string;
  transaction_id: string;
  title: string;
  meeting_date: string;
  meeting_phase: MeetingPhase;
  buyer_id: string | null;
  status: TranscriptionJobStatus;
  transcript: string | null;
  minutes_json: MeetingMinutesResult | null;
  meeting_log_id: string | null;
  stt_cost_krw: number;
  llm_cost_usd: number;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface MeetingMinutesResult {
  minutes: string;
  summary: string;
  key_issues: { title: string; description: string; priority: string }[];
  action_items: ActionItemDraft[];
  condition_assessment: {
    match_level: ConditionMatchLevel;
    notes: string;
  } | null;
  buyer_reaction: BuyerReaction | null;
}

export interface ActionItemDraft {
  title: string;
  description: string | null;
  assignee_name: string | null;
  assignee_email: string | null;
  due_date: string | null;
  priority: string;
}

export interface TranscriptionStartPayload {
  audio: File;
  title: string;
  meeting_date: string;
  meeting_phase: MeetingPhase;
  buyer_id?: string;
  attendees_json: string; // JSON 직렬화된 참석자 배열
}

export interface TranscriptionApprovalPayload {
  minutes: string;
  summary?: string;
  action_items?: ActionItemDraft[];
  condition_match?: string;
  condition_notes?: string;
  buyer_reaction?: string;
}
